packages/reactive-element/src/reactive-element.ts

```diff
@@ -651,7 +639,7 @@ export abstract class ReactiveElement
     if (options.state) {
       // Cast as any since this is readonly.
       // eslint-disable-next-line @typescript-eslint/no-explicit-any
-      (options as any).attribute = false;
+      options = { ...options, attribute: false };
```

Comment:
This is adding an object copy on a frequently called API, way too expensive to be worth the small gain in purity